### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by wyq730**
### Why:
* We need an overview page to show the overall map, and a algorithm running page to visualize the result.

### What:
* Finished basic UI of overview page & algorithm running page.

### CODE REVIEW

File: frontend/src/Components/About/About.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  render() {
+    return (
+      <div >
+        <Paperbase />
+      </div>
+    );
+  }
+}
+
+export default About;
```
Comment (hooshooo) [inline frontend/src/Components/About/About.js]: Please end up with a new line.
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: frontend/src/Components/Algorithms/Algorithm1.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <Content text={'Results'} compo={ <ResultCharts />} />
+        }
+        </div>
+        }/>
+      </div>
+    );
+  }
+}
+
+export default Algorithm1;
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm1.js]: same here

File: frontend/src/Components/Algorithms/Algorithm2.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        <Content text={'Scenario'} compo={<Scenario />}/> 
+        <br/><Content text={'Results'}/>
+        </div>
+        }/>
+      </div>
+    );
+  }
+}
+
+export default Algorithm2;
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm2.js]: same

File: frontend/src/Components/Algorithms/Algorithm3.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <div>
+        <Content text={'Scenario'} compo={<Scenario />}/> 
+        <br/><Content text={'Results'}/>        </div>
+        }/>
+      </div>
+    );
+  }
+}
+
+export default Algorithm3;
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm3.js]: same

File: frontend/src/Components/Algorithms/Algorithm4.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        <div>
+        <Content text={'Scenario'} compo={<Scenario />}/> 
+        <br/><Content text={'Results'}/>        </div>
+        }/>
+      </div>
+    );
+  }
+}
+
+export default Algorithm4;
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm4.js]: same

File: frontend/src/Components/Algorithms/Algorithm5.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        <Content text={'Scenario'} compo={<Scenario />}/> 
+        <br/><Content text={'Results'}/>            
+        </div>
+        }/>
+      </div>
+    );
+  }
+}
+
+export default Algorithm5;
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm5.js]: same

File: frontend/src/Components/Overview/Overview.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+              compo={ <OverviewMap /> }
+            />
+          }
+        />
+      </div>
+    );
+  }
+}
+
+export default Overview;
```
Comment (hooshooo) [inline frontend/src/Components/Overview/Overview.js]: same

File: frontend/src/Components/OverviewMap/OverviewMap.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+  onMouseOut = () => {
+    this.setState({ current: {
+      countyName: null,
+      totalEnergy: null,
+    }, isTooltipVisible: false });
+  }
+}
+
+export default OverviewMap;
```
Comment (hooshooo) [inline frontend/src/Components/OverviewMap/OverviewMap.js:197]: same

File: frontend/src/Components/OverviewMap/overviewMapStyled.js [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    var pctLower = 1 - rangePct;
+    var pctUpper = rangePct;
+    var color = {
+        r: Math.floor(lower.color.r * pctLower + upper.color.r * pctUpper),
+        g: Math.floor(lower.color.g * pctLower + upper.color.g * pctUpper),
+        b: Math.floor(lower.color.b * pctLower + upper.color.b * pctUpper)
+    };
+    return 'rgb(' + [color.r, color.g, color.b].join(',') + ')';
+    // or output as hex if preferred
+}
```
Comment (hooshooo) [inline frontend/src/Components/OverviewMap/overviewMapStyled.js]: same

File: frontend/src/Components/OverviewMap/sampleCounties.js [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+  "shasta": {
+    "totalEnergy": Math.random(),
+  },
+  "lassen": {
+    "totalEnergy": Math.random(),
+  },
+  "modoc":{
+    "totalEnergy": Math.random(),
+  }
+};
```
Comment (hooshooo) [inline frontend/src/Components/OverviewMap/sampleCounties.js]: same

File: frontend/src/Components/OverviewMap/sampleCounties.js [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+  },
+  "shasta": {
+    "totalEnergy": Math.random(),
+  },
+  "lassen": {
+    "totalEnergy": Math.random(),
+  },
+  "modoc":{
+    "totalEnergy": Math.random(),
+  }
```
Comment (hooshooo) [inline frontend/src/Components/OverviewMap/sampleCounties.js]: I think this is only for our demo, so I prefer not merging this into master branch.

File: frontend/src/Components/ResultChart/ResultChart.css [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+  display: inline-block;
+}
+
+.rv-xy-plot, .chart-grid {
+  display: inline-block;
+}
+
+.rv-xy-plot__axis__title.chart-title text {
+  font-size: 1.2rem;
+}
```
Comment (hooshooo) [inline frontend/src/Components/ResultChart/ResultChart.css]: same

File: frontend/src/Components/ResultChart/ResultChart.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+            position="end"
+            tickSize="10"
+          />
+        </XYPlot>
+      </div>
+    );
+  }
+}
+
+export default ResultChart;
```
Comment (hooshooo) [inline frontend/src/Components/ResultChart/ResultChart.js]: same

File: frontend/src/Components/ResultChart/ResultChart.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+import "./ResultChart.css"
+
+class ResultChart extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const data = [
+      {x: 1, y: 8},
```
Comment (hooshooo) [inline frontend/src/Components/ResultChart/ResultChart.js]: I think you suppose to get these data through a API call?

File: frontend/store.json [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+    },
+    {
+      "id": "love-freedom-aloneness-the-koan-of-relationships",
+      "writerId": "osho",
+      "title": "Love, Freedom, Aloneness: The Koan of Relationships",
+      "description": "",
+      "published": 2002
+    }
+  ]
+}
```
Comment (hooshooo) [inline frontend/store.json]: Is it something related to our project?

File: frontend/package.json [hunk 16]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "writers-blog",
```
Comment (hooshooo) [inline frontend/package.json:2]: I think we can change the name to something like `script-ui`?

File: frontend/.gitignore [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+package-lock.json
```
Comment (hooshooo) [inline frontend/.gitignore:25]: Seems it's not ignored in your commits.

File: frontend/src/Components/ResultChart/ResultCharts.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+
+function ResultCharts(props) {
+  return (
+    <>
+      <ResultChart />
+    </>
+  )
+}
+
+export default ResultCharts;
```
Comment (hooshooo) [inline frontend/src/Components/ResultChart/ResultCharts.js]: same

File: frontend/src/Components/Upload/Upload.js [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+  render() {
+    return (
+      <div >
+        <Base />
+      </div>
+    );
+  }
+}
+
+export default Upload;
```
Comment (hooshooo) [inline frontend/src/Components/Upload/Upload.js]: same

File: frontend/src/Components/Algorithms/Algorithm4.js [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+
+class Algorithm4 extends Component {
+  render() {
+    return (
+      <div >
+        <Base 
+        content={
+        <div>
+        <Content text={'Scenario'} compo={<Scenario />}/> 
+        <br/><Content text={'Results'}/>        </div>
```
Comment (hooshooo) [inline frontend/src/Components/Algorithms/Algorithm4.js]: I know little about react, but I think we are supposed to format the embedded html code as well.

**Review by hooshooo — CHANGES_REQUESTED** [review]
I think it would be better if it's merged into a non-master branch, such as develop.
As I said, I know little about react, so the only thing I can do is to check the format.
And could you please write some documentations about how to setup the environment or the structure of your code.

**Comment by wyq730** [comment]
Replaced by #54

**Pull request closed without merging**